app/models/manageiq/providers/infra_manager.rb

```diff
@@ -44,5 +44,24 @@ def self.merge_config_settings(cfg = VMDB::Config.new("vmdb"))
       path = [:ems, :ems_redhat, :service, :read_timeout]
       cfg.merge_from_template_if_missing(*path)
     end
+
+    def self.credential_validation(ems, auth_type, creds, revalidate)
+      ems.update_authentication(creds, :save => false)
+
+      ok, error_message = ems.authentication_check(auth_type, :save => revalidate)
+
+      raise MiqException::Error, error_message unless ok
+    end
+
+    def self.credential_validation_task(options)
```

Comment:
Yes that's what I'm looking at now actually, ems_cloud has their own create and validate_credentials methods so they don't run these in ems_common but I did notice it for ems_container types.